.changeset/hip-moons-play.md

```diff
@@ -0,0 +1,5 @@
+---
+'@emotion/memoize': minor
+---
+
+Migrate to TypeScript and emit declarations based on the code
```

Comment:
```suggestion
Source code has been migrated to TypeScript. From now on type declarations will be emitted based on that, instead of being hand-written.
```